contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,516 @@
+- tolerations that tolerates taints with key `other_key` that has a `NoExecute` effect.
+
+##### Fine-grain allowed tolerations
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    tolerations:
+      - match:
+        - keys: ["mykey"]
+          operators: ["Equal"]
+          values: ["value"]
+          effects: ["NoSchedule"]
+        - keys: ["other_key"]
+          operators: ["Exists"]
+          effects: ["NoExecute"]
+        action: allowed
+```
+
+Here we allow tolerations in the following forms:
+- tolerations that tolerates all `PreferNoSchedule` taints with any value.
+- tolerations that tolerates taints based on any key existence with effect `NoSchedule`.
+Also note that this SchedulingPolicy does not allow tolerating NoExecute taints.
+
+##### Coarse-grain allowed tolerations
+
+
+an empty list of toleration allows all types of tolerations:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    tolerations:
+      - match: []
+        action: allowed
+```
+
+Which is equivalent to:
+
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    tolerations:
+      - match:
+        - keys: []
+          operators: []
+          values: []
+          effects: []
+        action: allowed
+```
+
+
+
+### Priority classes
+
+Priority class usage can be regulated using fine-grain rules with `priorityClasseName` field.
+
+##### required
+
+this example requires a `priorityClass` that is either `high-priority` or `critical-job`
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    priorityClasseNames:
+      - match: ["high-priority","critical-job"]
+        action: required
+```
+
+
+
+##### Allow
+
+In this example, we only allow the `critical-job` priority.
+
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    priorityClasseNames:
+      - match: ["critical-priority"]
+        action: allowed
+```
+
+
+
+
+### Node Selector
+
+The `nodeSelector` field makes it possible to specify what nodeSelectors are required, allowed and denied. As for other components, `required` nodeSelectors are automatically considered as allowed.
+
+#### Examples
+
+##### Complete policy
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    nodeSelectors:
+      - match:
+        - beta.kubernetes.io/arch: ["amd64", "arm64"]
+        - team: []
```

Comment:
This match schema is different from label selector. I recommend using label selector schema here, instead of inventing something different.